### PR TITLE
Basic scaffolding for onboarding

### PR DESCRIPTION
**Pull request by jbirddog**
Adds a shim view to the frontend to allow changing the homepage, to a degree based on results from an onboarding process model that is called, if it exists. If the onboarding model has human tasks they will be handled normally. To support running the onboarding process `process_instance_create` and `process_instance_run` were refactored and a `process_instance_start` was added that does the two steps in one.

Likely may go back and add message support later, can either be the first check or done after the known location check that current exists fails.

### CODE REVIEW

File: spiffworkflow-frontend/src/routes/OnboardingView.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      if (onboarding.type === 'default_view') {
+        if (onboarding.value === 'my_tasks') {
+          return <MyTasks />;
+        }
+      } else if (
+        onboarding.type === 'user_input_required' &&
+        onboarding.process_instance_id &&
+        onboarding.task_id
+      ) {
+        console.log('HERE');
```
Comment (burnettk) [inline spiffworkflow-frontend/src/routes/OnboardingView.tsx:31]: console straggler

**Review by burnettk — APPROVED** [review]
awesome

File: spiffworkflow-frontend/src/routes/OnboardingView.tsx (hunk 1 above)
Comment (jbirddog) [reply spiffworkflow-frontend/src/routes/OnboardingView.tsx:31]: good catch

File: spiffworkflow-backend/src/spiffworkflow_backend/routes/process_instances_controller.py [hunk 2]
```diff
@@ -134,6 +149,14 @@ def process_instance_run(
     return make_response(jsonify(process_instance), 200)
 
 
+def process_instance_start(
```
Comment (jasquat) [inline spiffworkflow-backend/src/spiffworkflow_backend/routes/process_instances_controller.py:152]: Should we underscore this? That's what we've been doing when a method is called by a controller and not by the api. Helps to show what are actually api methods and what aren't.

**Review by jasquat — APPROVED** [review]
How hard would it be to add an integration test for this? Also, do we need a permission set for the "/onboarding" endpoint? Is it apart of BASIC permissions?

File: spiffworkflow-backend/src/spiffworkflow_backend/routes/process_instances_controller.py (hunk 2 above)
Comment (jbirddog) [reply spiffworkflow-backend/src/spiffworkflow_backend/routes/process_instances_controller.py:152]: No problems with that.

**Comment by jbirddog** [comment]
> How hard would it be to add an integration test for this? Also, do we need a permission set for the "/onboarding" endpoint? Is it apart of BASIC permissions?

I'm not sure honestly, i think it would be checking if the homepage did or did not do something different - maybe a cypress test? Don't have a confident answer but not opposed to a test either.

**Comment by jbirddog** [comment]
> Is it apart of BASIC permissions?

Added read for /onboarding to the basic permissions

**Comment by jbirddog** [comment]
Adding an integration test for when an onboarding model is present and when one isn't.

**Pull request merged**